code/modules/vehicles/mecha/mecha_mob_interaction.dm

```diff
@@ -184,9 +192,15 @@
 /obj/vehicle/sealed/mecha/container_resist_act(mob/living/user)
 	if(isAI(user))
 		var/mob/living/silicon/ai/AI = user
-		if(!AI.can_shunt)
-			to_chat(AI, span_notice("You can't leave a mech after dominating it!."))
-			return FALSE
+		if(!AI.linked_core)
+			to_chat(AI, span_userdanger("Inactive core destroyed. Unable to return."))
+			if(!AI.can_shunt || !AI.hacked_apcs.len)
+				to_chat(AI, span_warning(AI.can_shunt ? "No hacked APCs available." : "No shunting capabilities."))
```

Comment:
This is not safe code. 
When this resolves, it ends up being
`"<span class='warning'>"` + `1` + `"No hacked APCs available."` + `"</span>"`
Which runtimes because it's doing string + number. 

```suggestion
				to_chat(AI, span_warning("[AI.can_shunt ? "No hacked APCs available." : "No shunting capabilities."]"))
```